app/models/mixins/event_mixin.rb

```diff
@@ -36,10 +36,39 @@ def events_table_name(assoc)
     events_assoc_class(assoc).table_name
   end
 
+  def event_stream_filters
+    {
+      "EmsEvent".freeze => ems_event_filter,
+      "MiqEvent".freeze => miq_event_filter
+    }
+  end
```

Comment:
☝️ This is what is called by the UI to get 

```
{
 "MiqEvent"=>{"target_id"=>222, "target_type"=>"VmOrTemplate"},
 "EmsEvent"=>{"vm_or_template_id"=>222}
 }
```

Reply:
The ui asks a core object (e.g.: Vm) "what filter do I use to find events related to you?"

This is returned.

So we will need to find a way to make this a virtual attribute.

Reply:
yeah, I wanted to expose it in the API but it looks awful... enumerating all classes to get back {column => :id} for each of the event stream subclasses.  After cleaning up some of the ems event stuff in event stream, I'm hoping a better interface for getting this becomes more obvious. It's been tough not ripping this thing apart.